Taarafo.Portal.Web/Services/PostViews/PostViewService.cs

```diff
@@ -0,0 +1,45 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Taarafo.Portal.Web.Brokers.Loggings;
+using Taarafo.Portal.Web.Models.Posts;
+using Taarafo.Portal.Web.Models.PostViews;
+using Taarafo.Portal.Web.Services.Foundations.Posts;
+
+namespace Taarafo.Portal.Web.Services.PostViews
+{
+    public partial class PostViewService : IPostViewService
+    {
+        private readonly IPostService postService;
+        private readonly ILoggingBroker loggingBroker;
+
+        public PostViewService(
+            IPostService postService,
+            ILoggingBroker loggingBroker)
+        {
+            this.postService = postService;
+            this.loggingBroker = loggingBroker;
+        }
+
+        public ValueTask<List<PostView>> RetrieveAllPostViewsAsync() =>
+        TryCatch(async () =>
+        {
+            List<Post> posts =
+                await this.postService.RetrieveAllPostsAsync();
+
+            return posts.Select(AsPostView).ToList();
+        });
+
+        private static Func<Post, PostView> AsPostView =>
+            post => new PostView
+            {
+                Content = post.Content
+            };
```

Comment:
We still need the author too. And date. so people can see who posted and when it was posted right?

Reply:
And maybe also the Id so we can delete it later.

Reply:
@hassanhabib added the properties. Included UpdatedDate also since posts can be edited and we should show that as well in the view.